test/ERC6538Registry.t.sol

```diff
@@ -47,24 +67,296 @@ contract RegisterKeys is ERC6538RegistryTest {
   /// forge-config: default.fuzz.runs = 1
   /// forge-config: ci.fuzz.runs = 1
   /// forge-config: lite.fuzz.runs = 1
-  function testFuzz_NeverReverts(address caller, uint256 schemeId, bytes memory stealthMetaAddress)
-    external
-  {
+  function testFuzz_AlwaysSucceeds(
+    address caller,
+    uint256 schemeId,
+    bytes memory stealthMetaAddress
+  ) external {
     vm.prank(caller);
     registry.registerKeys(schemeId, stealthMetaAddress);
   }
 }
 
-contract RegisterKeysOnBehalf_Address is ERC6538RegistryTest {
-  function test_NotImplemented() external {
-    vm.expectRevert("not implemented");
+contract RegisterKeysOnBehalf is ERC6538RegistryTest {
+  function testFuzz_Erc712SignatureIsValid(
```

Comment:
Our tests names should describe the expected behavior in such a way that they generate a pseudo-specification for the contract when we run `scopelint spec`.

So, for example, here's what the `scopelint spec` output looks like for the Announcer:

```
Contract Specification: ERC5564Announcer
└── announce
    ├──  Emits Announcement Event
    └──  Always Succeeds
```

This output represents a spec for each method (there's only one) of this (very simple) contract. Does this make sense?

Here's the output for this method:

```
registerKeysOnBehalf
│   ├──  Erc712 Signature Is Valid
│   ├──  Erc1271 Signature Is Valid
│   ├──  Update Stealth Meta Address
│   ├──  Revert If: Erc712 Signature Is Not Valid
│   ├──  Revert If: Erc1271 Signature Is Not Valid
│   ├──  Revert If: Wrong Nonce
│   └──  Revert If: No Signature Is Provided
```

Several of these don't serve as description of the behavior being tested. This would not represent a spec for the contract. Looking at this first test, we'd want the output to say something like this:

```
registerKeysOnBehalf
│   ├── Sets The Stealth Meta Address For A New Registrant When Provided A Valid Erc712 Signature
│   ├── Emits A StealthMetaAddressSet Event For A New Registrant When Provided A Valid Erc712 Signature
etc...
```

This means we need to:

1. Rename many of these tests to produce useful spec descriptions
2. Create two independent unit tests for the event emission and the actual state update

Sorry for being stickler on this but given the nature of these contracts we want the test suite and spec to be exhaustive and clear!

Reply:
No worries! I made an attempt at making the test names as comprehensive as possible in this commit [3a5aa7c](https://github.com/ScopeLift/stealth-address-erc-contracts/pull/5/commits/3a5aa7ce082f3580802dcad5e30884c9a736e04a)